lib/common/http1client.c

```diff
@@ -457,6 +458,18 @@ static void on_head(h2o_socket_t *sock, const char *err)
         if (!(100 <= http_status && http_status <= 199 && http_status != 101))
             break;
 
+        if (http_status == 100 && client->_expect_100_continue) {
```

Comment:
Addressed in https://github.com/h2o/h2o/pull/3316/commits/fa2b8996341ae450a63c95896f593d6b93a29cd8

Talking about the name of the field, I feel `_expect_100_continue` is reasonable enough because it means just expecting 100, and turning it off obviously means `it is not expecting 100`, but I don't have strong opinion :)

Reply:
SG. But separately, please see my comment regarding if this logic should be specific to 100 and nothing else.